src/libs/routes/authmiddleware.ts

```diff
@@ -23,7 +23,7 @@ export default (module, permissionType) => (req, res, next) => {
             req.user = decode;
             res.locals.user = decode;
             console.log('User in request', decode);
-            if (!hasPermission(module, permissionType, decode.role)) {
+            if (hasPermission(module, permissionType, decode.role)) {
```

Comment:
hasPermission will return true if user has that permission, but here you return 403 if user has permissions for that module. Add "!" to fix this.